src/GraphQL/Utilities/Visitors/Custom/PatternMatchingVisitor.cs

```diff
@@ -37,21 +37,41 @@ public override void VisitInterfaceFieldArgumentDefinition(QueryArgument argumen
     {
         // look for @pattern directive applied to the field argument or input field, and if found, use Validate to set the validation method
         var applied = fieldArgOrInputField.FindAppliedDirective("pattern");
-        if (applied?.FindArgument("regex")?.Value is not string regex)
-            return null;
+        var value = applied?.FindArgument("regex")?.Value;
 
-        // if GlobalSwitches.DynamicallyCompileToObject is true, then compile
-        // the regex (or pull from the static cache if already compiled)
-        // (note that GlobalSwitches.DynamicallyCompileToObject is false for AOT schemas and scoped schemas)
-        var regexObject = new Regex($"^{regex}$",
-            GlobalSwitches.DynamicallyCompileToObject ? RegexOptions.Compiled : RegexOptions.None);
+        var regexObject = value switch
+        {
+            // if GlobalSwitches.DynamicallyCompileToObject is true, then compile
+            // the regex (or pull from the static cache if already compiled)
+            // (note that GlobalSwitches.DynamicallyCompileToObject is false for AOT schemas and scoped schemas)
+            string regex => new Regex($"^{regex}$",
+                GlobalSwitches.DynamicallyCompileToObject ? RegexOptions.Compiled : RegexOptions.None),
+            Regex r => r,
+            _ => null
```

Comment:
I think it should throw if the value is of not supported type rather than silently skipping the directive (same is in the current version)